tests/test_typing.py

```diff
@@ -0,0 +1,40 @@
+# This is test file for typing,
+# No automatic testing is used at the moment. Just use your type checker and see if it works.
```

Comment:
Just being curious here: isn't there a way to assert typing-related things in pytest ? We can even consider using `typing-inspect` or another such package if it makes sense..

Alternately we could add a `mypy` step in the nox file, scanning precisely this test file ?

Reply:
I remember few days ago seeing article about testing typing in python packages, but couldn't find it at the moment.

We cannot just run `mypy` or `pyright` in CI, becouse it will exit with errors, and we want to be sure that it fails where we expect it to fail.

I think, our best shot is to run `mypy` and `pyright` using subprocess in `pytest`, and compare it's output with what we're expecting. I could try to do this in next PR or in this, what do you think?

But before that i want to think a little about other options, and maybe see what others are doing.

Reply:
As for `typing-inspect`, i don't think it could be used in this case. Sure it could check something, but i'd want to test exact cases.

Reply:
I see

Reply:
Actually, i found something.

django-types are running `pyright` in subprocess, and parsing it's messages

https://github.com/sbdchd/django-types/blob/main/tests/pyright/base.py

and, mypy could be called from pytest using it's api:

https://mypy.readthedocs.io/en/stable/extending_mypy.html

I tried using mypy, but strumbled upon a bug which crashes mypy. Now i'm trying to fix it or make minimal example to report bug in mypy.

I think, merging this could actually crash mypy for someone who uses it.

https://github.com/python/mypy/issues/8645

They already have similar issues with other code.

Reply:
Oh, nevermind. It was crashing becouse it has cached some data from previous version in my environment. It's working fine. Just don't support those ParamSpecs properly yet.

So, i think, if we want to add type testing, we could use pyright just like django-types. (But maybe that's for another PR)